README.md

```diff
@@ -251,6 +257,128 @@ Here's an example of the typical output to the console:
 15
 ```
 
+#### Generate a waveform
+
+```js
+var outPin = 17;
+
+var Gpio = require('pigpio').Gpio,
+  out = new Gpio(outPin, {
+    mode: Gpio.OUTPUT
+  });
+  
+(function () {
+  var waveform = [];
+  
+  var x = 0;
+  for (x=0; x < 10; x++) {
+    if (x % 2 == 0) {
+      waveform[x] = { gpioOn:(1 << outPin), gpioOff:0, usDelay:20 };
+    } else {
+      waveform[x] = { gpioOn:0, gpioOff:(1 << outPin), usDelay:20 };
+    }
+  }
+  
+  for (x=10; x < 20; x++) {
+    if (x % 2 == 0) {
+      waveform[x] = { gpioOn:(1 << outPin), gpioOff:0, usDelay:30 };
+    } else {
+      waveform[x] = { gpioOn:0, gpioOff:(1 << outPin), usDelay:30 };
+    }
+  }
+  
+  for (x=20; x < 30; x++) {
+    if (x % 2 == 0) {
+      waveform[x] = { gpioOn:(1 << outPin), gpioOff:0, usDelay:20 };
+    } else {
+      waveform[x] = { gpioOn:0, gpioOff:(1 << outPin), usDelay:20 };
+    }
+  }
+  
+  Gpio.waveClear();
+  
+  Gpio.waveAddGeneric(waveform.length, waveform);
+  
+  var waveId = Gpio.waveCreate();
+
+  if (waveId >= 0)
+  {
+    Gpio.waveTxSend(waveId, Gpio.WAVE_MODE_ONE_SHOT);
+  }
+
+  while (Gpio.waveTxBusy()) {}
+  
+  Gpio.waveDelete(waveId);
+  
+}());
+```
+#### Sending a wavechain
+
+```js
+      var outPin = 17;
+
+      var Gpio = require('pigpio').Gpio,
+              out = new Gpio(outPin, {
+                mode: Gpio.OUTPUT
+              });
+      var chain = [];
+      var waveform = [];
+      var waveId;
+      var x = 0;
+      for (x=0; x < 10; x++) {
+        if (x % 2 == 0) {
+          chain[x] = { gpioOn:(1 << outPin), gpioOff:0, usDelay:20 };
+        } else {
+          chain[x] = { gpioOn:0, gpioOff:(1 << outPin), usDelay:20 };
+        }
+      }
+      Gpio.waveClear();
+      Gpio.waveAddGeneric(waveform.length, waveform);
+      waveId = Gpio.waveCreate();
+      
+      chain.push(255)
+      chain.push(1)
+      chain.push(6)
+      chain.push(0);
+      
+      var bufchain = Buffer.from(chain)
+      Gpio.waveChain(bufchain, bufchain.length);
+      while(Gpio.waveTxBusy()){
+      }
+      Gpio.waveDelete(waveId);
+
+```
+#### Adding a waveform representing serial data
```

Comment:
Have you verified that the example "Adding a waveform representing serial data" works?
I haven't tested it but think it will terminate before the serial data is sent.
Please verify and confirm that the example works.
The code should not be indented.

Reply:
You are correct.
Fixed in f716731